src/asl.ts

```diff
@@ -1807,31 +2019,48 @@ export class ASL {
           }
         }
 
-        return this.evalExpr(
-          objParamExpr,
-          (_, { normalizeOutputToJsonPath }) => {
-            const objectPath = normalizeOutputToJsonPath().jsonPath;
+        return this.evalExprToJsonPath(objParamExpr, (output) => {
+          const objectPath = output.jsonPath;
 
+          return {
+            Type: "Pass",
+            Parameters: {
+              // intrinsic functions cannot be used in InputPath and some other json path locations.
+              // We compute the value and place it on the heap.
+              "string.$":
+                isPropAccessExpr(expr.expr) &&
+                isIdentifier(expr.expr.name) &&
+                expr.expr.name.name === "stringify"
```

Comment:
Maybe pull this out into a helper, `isJsonParse` and `isJsonStringify`. They're used earlier too and it's a bit verbose